enhancements/sdk-kubernetes-1.20.md

```diff
@@ -0,0 +1,248 @@
+---
+title: Support Kubernetes 1.19 in Operator SDK
+authors:
+  - "@bharathi-tenneti"
+reviewers:
+  - "@estroz"
+  - "@camilamacedo86"
+  - "@jmrodri"
+approvers:
+  - "@joelanford"
+  - "@camilamacedo86"
+  - "@jmrodri"
+creation-date: 2020-09-25
+last-updated: 2020-09-25
+status: provisional
+see-also:
+  - "https://github.com/kubernetes/sig-release/tree/master/releases/release-1.20"
+replaces: []
+superseded-by: []
+---
+
+# Support Kubernetes 1.19 in Operator SDK
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] ~~Graduation criteria for dev preview, tech preview, GA~~ (N/A)
+
+## Open Questions
+
+
+## Summary
+
+Operator SDK consists of many components (project scaffolding, OLM integrations, scorecard, helm and
+ansible operators, and a Go library for operator authors) that have interactions with Kubernetes
+clusters. The goal of this enhancement is to upgrade ALL components of Operator SDK to be compatible
+with Kubernetes 1.20.
+
+
+## Motivation
+
+Operator SDK needs to support the latest features of Kubernetes since that operator developers expect to be able to use them shortly after the release is made.
+
+## Dependencies
+
+[external] Require controller-runtime release that supports 1.20
+[external] Require Helm release that supports 1.20
+[external] Require kubebuilder release that scaffolds projects using 1.20
+[internal] Require operator-framework/api release that supports 1.20
+
+
+### Goals
+
+- Operator SDK’s dependencies are upgraded to use Kubernetes 1.20.
+- All e2e tests run during Operator SDK CI are upgraded to test against Kubernetes 1.20.
+- New projects scaffolded with Operator SDK are upgraded to use Kubernetes 1.20 dependencies.
+- New projects scaffolded with Operator SDK should generate v1 CRDs by default.
+
+
+### Non-Goals
+
+- Feature development and bug fixes that are not directly related to Kubernetes 1.20 support. The
+  rubric here is, if after bumping `go.mod` dependency versions:
+  - Will the test suite fail CI without this change?
+  - Will ignoring this change cause a regression?
+
+  If the answer to these questions is no, then it is out of scope for this enhancement.
+  - Unrelated bug fixes can be done before or after this enhancement is implemented.
+  - Integration of new features introduced in 1.20 will be done in follow-on PRs.
+
+- SDK Go version bump to 1.15
+  - Risk: users are often slow to upgrade go versions, but upstream deps might use go 1.15 features.
+  - Mitigation: we will only bump go version to 1.15 if there are breaking changes (go 1.15 features) introduced by Kubernetes 1.20.
+
+## Proposal
+
+### User Stories
```

Comment:
The following are the tasks required to accomplished the goa, in this way, IMO should be in the Graduation Criteria. See that the User stories are optional so we can do not fill it if it is the case. 

However, if we would add the info in this section it would be about the value that these changes/goals bring to end-users. And then, should be in the format `As a <role> I can <capability>, so that <receive benefit>`  See:

> A user story is a tool in Agile software development used to capture a description of a software feature **from a user's perspective.** The user story describes the type of user, what they want and why. A user story helps to create a simplified description of a requirement.
ref: https://searchsoftwarequality.techtarget.com/definition/user-story

> User stories are short, simple descriptions of a feature told from the perspective of the person who desires the new capability, usually a user or customer of the system. They typically follow a simple template:
>
> As a < type of user >, I want < some goal > so that < some reason >.
Ref; https://www.mountaingoatsoftware.com/agile/user-stories

> User stories are often written from the perspective of an end user or user of a system
> As a <role> I can <capability>, so that <receive benefit>
Ref: https://en.wikipedia.org/wiki/User_story